include/maths/CDataFrameAnalysisStateInterface.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+#ifndef INCLUDED_ml_maths_CDataFrameAnalysisInterface_h
+#define INCLUDED_ml_maths_CDataFrameAnalysisInterface_h
+
+#include <maths/ImportExport.h>
+
+#include <cstdint>
+
+namespace ml {
+namespace maths {
+
+//! \brief Interface class for collecting data frame analysis job statistics owned
+//! by the maths module.
+class MATHS_EXPORT CDataFrameAnalysisStateInterface {
```

Comment:
I'm wondering what the rationale is for having `CDataFrameAnalysisStateInterface` and `CDataFrameAnalysisState`, i.e. I'm not sure the extra layer adds anything and `progressCallback` etc could be pushed down. If it is to stub out functionality for the tests. I'd declare a version in the test library which derives from `CDataFrameAnalysisState` which does this.

Reply:
As discussed, I want the `maths` to own their own reference to avoid dependency to `api`